File: src/Enterprise/C#/Example2.cs
using System;

// Abstract Product
public interface Database {
    void Connect();
    void Query();
}

// Concrete Product
public class Postgres : Database {
    public void Connect() {
        Console.WriteLine("Connecting to PostgreSQL");
    }
    public void Query() {
        Console.WriteLine("Querying PostgreSQL");
    }
}

public class MySQL : Database {
    public void Connect() {
        Console.WriteLine("Connecting to MySQL");
    }
    public void Query() {
        Console.WriteLine("Querying MySQL");
    }
}

// Abstract Factory
public interface DatabaseFactory {
    Database CreateDatabase();
}

// Concrete Factory
public class PostgresFactory : DatabaseFactory {
    public Database CreateDatabase() {
        return new Postgres();
    }
}

public class MySQLFactory : DatabaseFactory {
    public Database CreateDatabase() {
        return new MySQL();
    }
}

public class Example2 {
    public static void UseDatabase(DatabaseFactory factory) {
        var db = factory.CreateDatabase();
        db.Connect();
        db.Query();
    }

    public static void Main(string[] args) {
        UseDatabase(new PostgresFactory());
        UseDatabase(new MySQLFactory());
    }
}

File: src/Enterprise/Java/Example2.java
// Abstract Product
interface Database {
    void connect();
    void query();
}

// Concrete Product
class Postgres implements Database {
    @Override
    public void connect() {
        System.out.println("Connecting to PostgreSQL");
    }

    @Override
    public void query() {
        System.out.println("Querying PostgreSQL");
    }
}

class MySQL implements Database {
    @Override
    public void connect() {
        System.out.println("Connecting to MySQL");
    }

    @Override
    public void query() {
        System.out.println("Querying MySQL");
    }
}

// Abstract Factory
interface DatabaseFactory {
    Database createDatabase();
}

// Concrete Factory
class PostgresFactory implements DatabaseFactory {
    @Override
    public Database createDatabase() {
        return new Postgres();
    }
}

class MySQLFactory implements DatabaseFactory {
    @Override
    public Database createDatabase() {
        return new MySQL();
    }
}

public class Example2 {
    public static void useDatabase(DatabaseFactory factory) {
        Database db = factory.createDatabase();
        db.connect();
        db.query();
    }

    public static void main(String[] args) {
        useDatabase(new PostgresFactory());
        useDatabase(new MySQLFactory());
    }
}
